### DISCUSSION TITLE
how to search the checkbox status?

### DISCUSSION THREAD

**Discussion by leeyaunlong** [TriliumNext/Trilium#6332] [Q&A] [answered]
Hi,
In editor, I create some checkout for todo (checked, unchecked), 
how to search the checkbox status with search function?

current , i can search the "checkbox" or "checked" to get the notes with checkbox.(I think that use full search the html code input type=checkbox , checked="checked" to implement it.)
but cannot query the checkbox status "unchecked".

**Comment by zadam** [comment]
Yeah, HTML checkboxes are not really designed for searchability. If I wanted to be able to search, I would use promoted attribute with type checkbox instead.

**Comment by leeyaunlong** [comment]
> Yeah, HTML checkboxes are not really designed for searchability. If I wanted to be able to search, I would use promoted attribute with type checkbox instead.

In the common cases, the note will include many tasks.
if we add a custom property (uncheck) in html, and with checked event to add /remove it.
I think we can improve trilium todo system.
> query all include todo checkbox : checkbox
> query uncheck : uncheck
> query checked : checkbox checked

**Comment by brownterryn** [comment]
FWIW I was able to search for unchecked checkboxes with

```
#!notToDo and note.rawContent %= '="checkbox"(?!.*checked)' and note.title != "Day template"
```
Explanation:
`#!notToDo` - label used to exclude things that need excluding
`note.title != "Day template"` - can't use `#notToDo` here because it's template, not attribute, inheritence

**Reply by MartinEnders** [reply]
Thank you for this search - it is exactly what I was looking for.

**Comment by karldonteljames** [comment]
How do I use this? I'm trying to get a note that collects all my todo lists (regardless of html or markdown) and show them in one place. In an ideal world, it would be great to see a date next to them, or have them grouped by their original note. I've spent a few days trying to utilise a JS frontend script, and a render note, but just seem to be having much luck.

Thanks

**Reply by eliandoran** [reply, marked as answer]
You press the "Search Notes" button in the launch bar and paste the search query provided by @brownterryn and then run the search.

You can save this search by pressing "Save to note".

You can even add `#expanded` to expand all the results by default or add a `#viewType=grid` to display them as a grid instead.

**Comment by karldonteljames** [comment]
Thanks, This is close enough to what i need. Just wondered if it is possible to have this note displayed without the search query being displayed, i.e. like a render note, or have the quesry run and hide automatically when that note is opened?
